force-app/main/classes/EventSink.cls

```diff
@@ -24,9 +27,12 @@ public class EventSink implements EventSinkInterface {
   }
 
   public void sinkGeneric(String kind, String raw) {
-    Integer count = [SELECT COUNT() FROM EventData__c];
-
-    if (count >= this.maxEvents) {
+    // Preserve SOQL queries (each transaction only gets 100) by getting count once and comparing to updated events count
+    if (existingEventsCount == null) {
```

Comment:
TODO: These instances all shared `EventData__c` (?). If so, it's not safe to query the existing events count once because we could have a race condition where multiple instances think it's safe to add new events, but it is not.